internal/bootstrap/bootstrap_plain_git.go

```diff
@@ -287,10 +288,28 @@ func (b *PlainGitBootstrapper) ReconcileSyncConfig(ctx context.Context, options
 	if err != nil && err != git.ErrNoStagedFiles {
 		return fmt.Errorf("failed to commit sync manifests: %w", err)
 	}
+
 	if err == nil {
 		b.logger.Successf("committed sync manifests to %q (%q)", b.branch, commit)
 		b.logger.Actionf("pushing sync manifests to %q", b.url)
-		if err = b.git.Push(ctx, b.caBundle); err != nil {
+		err = b.git.Push(ctx, b.caBundle)
+		if err != nil {
+			if strings.HasPrefix(err.Error(), gogit.ErrNonFastForwardUpdate.Error()) {
+				b.logger.Warningf("git conflict detected, retrying with a fresh clone")
+				if err := os.RemoveAll(b.git.Path()); err != nil {
+					return fmt.Errorf("failed to remove tmp dir: %w", err)
+				}
+				if err := os.Mkdir(b.git.Path(), 0700); err != nil {
```

Comment:
```suggestion
				if err := os.Mkdir(b.git.Path(), 0o700); err != nil {
```
since we've started using octal notation in other repos.

Reply:
I changed it, but Go itself doesn't not uses octal notation, why are we doing this?

https://github.com/golang/go/blob/334a591a3f4d868368913328b3e81ddf5b0f46fa/src/os/tempfile.go#L94

Reply:
Does the octal notation works on Windows?

Reply:
I don't have a windows environment, but based on this https://github.com/golang/go/blob/96567fb3ccdc8d8c052606112db7b39abd725b59/src/os/types.go#L23-L28 I'd assume it works.
> // A FileMode represents a file's mode and permission bits.
> // The bits have the same definition on all systems, so that
> // information about files can be moved from one system
> // to another portably. Not all bits apply to all systems.

Reply:
Thanks @pjbgf for the explanations